### PR TITLE
Parallel thumbnail fetching

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

- [x] Tests written, or not not needed

### CODE REVIEW

File: app/src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 1]
```diff
@@ -462,19 +462,27 @@ public ThumbnailGenerationTask(ImageView imageView, FileDataStorageManager stora
...
 
-        public ThumbnailGenerationTask(ImageView imageView, FileDataStorageManager storageManager,
-                                       User user, List<ThumbnailGenerationTask> asyncTasks,
-                                       boolean gridViewEnabled)
+        public ThumbnailGenerationTask(ImageView imageView,
+                                       FileDataStorageManager storageManager,
+                                       User user,
+                                       List<ThumbnailGenerationTask> asyncTasks,
+                                       boolean gridViewEnabled,
+                                       String remoteId)
```
Comment (AlvaroBrey) [inline app/src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java:470]: Parameter should be named `imageKey` if the field and the getter are named `imageKey`.

File: app/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java [hunk 2]
```diff
@@ -620,13 +622,19 @@ private static void setThumbnail(OCFile file,
                 } else {
                     // generate new thumbnail
                     if (ThumbnailsCacheManager.cancelPotentialThumbnailWork(file, thumbnailView)) {
+                        for (ThumbnailsCacheManager.ThumbnailGenerationTask task : asyncTasks) {
+                            if (file.getRemoteId() != null && task.getImageKey() != null && file.getRemoteId().compareTo(task.getImageKey()) == 0) {
```
Comment (AlvaroBrey) [inline app/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java:626]:
```suggestion
                            if (file.getRemoteId() != null && task.getImageKey() != null && file.getRemoteId().equals(task.getImageKey())) {
```

**Review by AlvaroBrey — APPROVED** [review]

**Comment by AlvaroBrey** [comment]
@tobiasKaminsky this change didn't go into 3.20.0 afaik, should we backport it to 3.20.1?

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky this change didn't go into 3.20.0 afaik, should we backport it to 3.20.1?

Yes 👍

**Pull request merged**